031-tls-encrypt-the-kafka-connect-rest-api.md

```diff
@@ -0,0 +1,172 @@
+# TLS encrypting the Kafka Connect REST API
+
+This proposal adds the option to encrypt communication with the Kafka Connect REST API by enabling TLS encryption on the Kafka Connect REST listener.
+
+## Current situation
+
+Currently, instances of Kafka Connect that are deployed by the Strimzi operators are configured with the default REST API endpoint settings.
+This means that the Kafka Connect REST API endpoint uses HTTP on port 8083 and that the Strimzi `KafkaConnect`, `KafkaConnector` and `KafkaMirrorMaker2` operators make unsecured REST client calls based on this default configuration.
+
+The default network policies created by the Strimzi operators restrict incoming REST API calls to only allow access from the operator pod.
+Users can define further network policies to override the default policy and allow wider access.
+
+## Motivation
+
+There is currently no way to TLS encrypt the Kafka Connect REST API, leaving the Connect REST listener as perhaps the last key endpoint that cannot currently be encrypted using Strimzi.
+
+## Proposal
+
+This proposal adds REST api configuration options to the `KafkaMirrorMaker2` and `KafkaConnect` CRDs.
+
+A new field is added to the `KafkaConnect` and `KafkaMirrorMaker2` specs named `restListeners`, which allows users to configure the Kafka Connect configuration to have:
+  - a single HTTP listener on port 8083
+  - a single encrypted HTTPS listener on port 8443
+  - both an HTTP listener on port 8083 and an encrypted HTTPS listener on port 8443
+  
+This enables users to access the REST API with TLS if required.
+The `KafkaConnector`, `KafkaConnect` and `KafkaMirrorMaker2` operators will use the TLS encrypted listener when it is enabled, even if an unencrypted listener is present.
```

Comment:
I touched on it below as well -> but it should be probably added here as well. The TLS is also used internally between Connect nodes to forward REST requests etc. So it is not just the user and the operator but also the Connect node themself.